bin/shutter

```diff
@@ -464,9 +464,20 @@ sub STARTUP {
 	$window->set_default_size(-1, 500);
 
 	#UPDATE WINDOW LIST (signal is connected when GUI is loaded)
-	my $wnck_screen = Wnck::Screen::get_default();
-	$wnck_screen->force_update if $wnck_screen;
-	my $x11_supported = $wnck_screen ? 1 : 0;
+	my $x11_supported;
+	if ($ENV{XDG_SESSION_TYPE} eq "wayland") {
+		$x11_supported = 0;
+	} elsif ($ENV{XDG_SESSION_TYPE} eq "x11") {
+		$x11_supported = 1;
+	} else {
+		print "ERROR: Unable to determine XDG_SESSION_TYPE!\n";
```

Comment:
I don't know whether we want to hard depend on XDG. Maybe default it to "assume supported" instead?

As is, this PR will break users who start X11 server manually and set `DISPLAY` variable to point at it, there there's nothing involved which would set the XDG vars, like KDE or GNOME. For example, this can happen when running shutter via `ssh -X`

Reply:
That's a good idea, will do! Is there a possible constellation where XDG_SESSION_TYPE won't give "wayland" on a Wayland setup?

Reply:
No clue

Reply:
I don't have much clue about Wayland in general, other than what I investigated for the dbus xdg portal prototype I wrote